_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,434 @@
+You have to use some other tool to assign the partitions from the broker to be removed, to the remaining brokers.
+The most convenient tool for this job is the Kafka partition reassignment tool.
+
+<!--more-->
+
+##  Kafka reassignment partition tool
+
+The `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+While using the tool, you have to provide it with either of these two JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` that we can use directly or modify further.
+
+- The `reassignment.json` file is a configuration file that is used during the partition reassignment process. The reassignment partition tool will generate a proposal `reassignment.json` file based on a `topics.json` file. You can change the `reassignment.json` file as per your requirement and use it.
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+- You can reassign partitions between the brokers. For example, it can be used when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+- You can increase the number of partitions / replicas which can help with increasing the throughput of topics.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers often leads to additional interbroker network traffic, in addition to the normal traffic required for replication. To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
+This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes per second, for example `--throttle 5000000` sets the limit to 5 MB/s.
+
+Throttling might cause the reassignment to take longer to complete.
+
+1. If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+2. If the throttle is too high, the overall health of the cluster may be impacted.
+
+You can use the `kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)` metric (which with the default spec.metrics settings in your Kafka CR would appear be scraped by Prometheus as `kafka_server-fetcherlagmetrics_consumerlag`) to observe how far the followers are lagging behind the leader
```

Comment:
Multiple things here ...

1. there is no `spec.metrics` settings in the Kafka CR. The field is `spec.kafka.metricsConfig`.
2. there is no default for it. I mean, if it's missing no metrics are exposed. If the corresponding ConfigMap has an "empty" JMX exporter configuration then all metrics are exposed.
3. I guess you are referring to the example we have in the `kafka-metrics.yaml`. You should explicitly refers to it or to the pattern used for exposing such a metric (I would probably prefer to refer to the example).